### PR TITLE
Allow manipulating state before persistence

### PR DESCRIPTION
**Pull request by DonnieWest**
@developit spitballing here, but would something like this solve your use case that you mentioned?

Really liking how much more flexible something like this is over a whitelist or blacklist of object keys. Open to ideas from anyone on implementation or how to name some of this stuff :)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -69,14 +69,17 @@ import localStorageAdapter from 'unissist/integrations/localStorageAdapter';
...
   version: 1,
   debounceTime: 100,
+  // called when version is updated. Defaults to dropping the store 
   migration: (oldState, oldversion) => ({ /* new state */ }),
+  // takes in the current state and returns the state to be persisted
+  map: state => ({ /* new persisted state shape */ })
 }
 
-persistStore(store, adapter);
+persistStore(store, adapter, conf);
```
Comment (developit) [inline README.md]: `config`?

File: src/index.js [hunk 2]
```diff
@@ -13,7 +13,12 @@ export default function persistStore(store, adapter, conf) {
   conf = conf || {};
   let version = conf.version || 1,
     debounceTime = conf.debounceTime || 100,
-    migration = conf.migration;
+    migration = conf.migration,
+    map =
+      conf.map ||
+      function(state) {
```
Comment (developit) [inline src/index.js]: Can use the builtin identity function for this:

```js
map = conf.map || Object;
```

File: src/index.js [hunk 3]
```diff
@@ -37,7 +42,7 @@ export default function persistStore(store, adapter, conf) {
     unsubscribe = store.subscribe(function() {
       if (!timer)
         timer = setTimeout(function() {
-          adapter.setState(store.getState());
+          adapter.setState(map(store.getState()));
```
Comment (developit) [inline src/index.js]: Could even inline the identity fallback here:

```js
adapter.setState((conf.map || Object)(store.getState()));
```

**Review by developit — COMMENTED** [review]
Just some notes, otherwise looks good!

**Comment by DonnieWest** [comment]
@developit thanks!

**Pull request merged**